### PR TITLE
Renovate most deps and code

### PR DESCRIPTION
**Pull request by Uzlopak**
This PR updates the deps and code. 
I just did not touch yargs.

### CODE REVIEW

**Comment by Uzlopak** [comment]
@shogunpanda

Do you know who could review and merge this?

**Comment by ShogunPanda** [comment]
I can!
I'm on phone now, but I'll take a look soon!

**Comment by Trott** [comment]
Hey, @nodejs/tsc: It looks like @indutny is the only owner of this repo. Which means only @indutny and members of @nodejs/tsc can merge code here. Might we want to add a team? (I don't know if @nodejs/llhttp or @nodejs/llnode would be appropriate?) Regardless, can someone review (and hopefully merge) this?

**Review by anonrig — APPROVED** [review]

**Comment by anonrig** [comment]
I'm ok with landing this but I'll give some time to other TSC members for a chance to review.

File: package.json [hunk 1]
```diff
@@ -33,15 +33,15 @@
...
-    "mocha": "^7.1.1",
-    "ts-node": "^8.8.1",
-    "tslint": "^5.20.1",
-    "typescript": "^3.9.6"
+    "@stylistic/eslint-plugin": "^1.5.4",
+    "@types/node": "^20.11.16",
+    "@typescript-eslint/eslint-plugin": "^6.20.0",
+    "@typescript-eslint/parser": "^6.20.0",
+    "eslint": "^8.56.0",
+    "ts-node": "^10.9.2",
```
Comment (benjamingr) [inline package.json:41]: nit: We can drop ts-node and use `--experimental-strip-types` (and still keep typescript in order to have type checking)

**Review by benjamingr — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Uzlopak) [reply package.json:41]: I would like, but I cant use  --experimental-strip-types with latest node 22 which is available via nvm.

**Pull request merged**